File: app/src/main/java/in/woloo/www/period_tracker/circular_calender/PERIOD_TYPES.java
package in.woloo.www.period_tracker.circular_calender;
import android.graphics.Color;

public enum PERIOD_TYPES {
    PERIOD("Normal",0,"#f7eb30","#F7C330"),
    MENSTRUATION("Menstruation",1,"#f8646b","#fee8e9"),
    OVULATION("Ovulation",2,"#2abdc4","#dff5f6"),
    PREGNANCY("High Fertility",3,"#ed8524","#fcedde");



    private String stringValue;
    private int intValue;
    private int intColor;
    private int intInnerColor;

    private PERIOD_TYPES(String toString, int value, String colorString, String innerColorString){
        stringValue = toString;
        intValue = value;
        intColor = Color.parseColor(colorString);
        intInnerColor = Color.parseColor(innerColorString);
    }

    @Override
    public String toString(){
        return stringValue;
    }

    public int color(){
        return intColor;
    }
    public int innerColor(){
        return intInnerColor;
    }
}

File: app/src/main/java/in/woloo/www/more/period_tracker/circular_calender/PERIOD_TYPES.kt
package `in`.woloo.www.more.period_tracker.circular_calender

import android.graphics.Color
import `in`.woloo.www.R

enum class PERIOD_TYPES(
    private val stringValue: String,
    private val intValue: Int,
    colorString: String,
    innerColorString: String
) {
    PERIOD("Normal", 0, "#FAED10", "#F7C330"),
    MENSTRUATION("Menstruation", 1, "#F1779B", "#fee8e9"),
    OVULATION("Ovulation", 2, "#B0C8E8", "#dff5f6"),
    PREGNANCY("High Fertility", 3, "#DFA7CC", "#fcedde");


    private val intColor = Color.parseColor(colorString)
    private val intInnerColor = Color.parseColor(innerColorString)

    override fun toString(): String {
        return stringValue
    }

    fun color(): Int {
        return intColor
    }

    fun innerColor(): Int {
        return intInnerColor
    }
}
